packages/xstate-store/test/store.test.ts

```diff
@@ -766,6 +803,45 @@ it('can be created with a logic object', () => {
   store.getSnapshot().context.count satisfies string;
 });
 
+it('should not trigger update if the snapshot is the same', () => {
```

Comment:
note this diverges from XState too:
```ts
import { createActor, createMachine } from "xstate";

const m = createMachine({
  on: {
    FOO: {
      actions: () => {},
    },
  },
});

const a = createActor(m).start();

let s = a.getSnapshot();

a.subscribe((_s) => {
  console.log("next", s === _s, _s);
  s = _s;
});

a.send({ type: "FOO" });
a.send({ type: "FOO" });
a.send({ type: "FOO" });
```

With the abo
<img width="833" alt="Screenshot 2025-07-07 at 08 56 07" src="https://github.com/user-attachments/assets/8edc7123-13a5-4e4a-b874-c31230faa063" />
ve we get:

Reply:
What do you think about this behavior? If only actions/effects occur (invisible to the state, at least right now), then the state is going to be exactly the same. But I guess that's the responsibility of the consumer to "filter"

Reply:
Yes, I think in XState v6, the purpose of the observer is to observe snapshot _changes_, so notifying on the same snapshot isn't useful.